### ISSUE TITLE
Trainer doesn't calculate eval loss for models with lm_labels parameter

### ISSUE THREAD

**Issue by patil-suraj** [huggingface/transformers#4361]
# 🐛 Bug
The trainer doesn't calculate evaluation loss for models with `lm_labels` parameters in the `forward` method. This line https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L682 is the source of the bug
`has_labels = any(inputs.get(k) is not None for k in ["labels", "masked_lm_labels"]) `
loss is only calculated when `has_labels` is `True` and its only `True` for models with `labels` and `masked_lm_labels` parameters and not for `lm_labels`

## Information

Model I am using (Bert, XLNet ...): T5

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

Use any model with `lm_labels` parameter in the `forward` method, ex T5 or BART and evaluate using trainer

## Expected behavior
Evaluation loss should be calculated for models with `lm_labels` parameter as well.

## Environment info

     
- `transformers` version: 2.9.1
- Platform: Linux-4.19.104+-x86_64-with-Ubuntu-18.04-bionic
- Python version: Python 3.6.9
- PyTorch version (GPU?):  1.5.0+cu101
- Tensorflow version (GPU?): Not applicable 
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No
